### PR TITLE
Acoustic examples merged

### PR DESCRIPTION
**Pull request by mloubout**
Small PR  that merged constant example as an argument to avoid useless duplicate.

Also avoids segfaults that were coming from `nbpml=0` in `constant_example` as the rec interpolation was trying to access a point outside the domain.

@FabioLuporini , it runs fine with GCC(4.9) and any `dse` so not sure what to say about #328

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -83,7 +83,7 @@ script:
       python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5;
       python examples/seismic/benchmark.py test -P acoustic -a;
       python examples/seismic/acoustic/acoustic_example.py --full;
-      python examples/seismic/acoustic/constant_example.py --full;
+      python examples/seismic/acoustic/acoustic_example.py -c;
```
Comment (FabioLuporini) [inline .travis.yml]: should we keep both ``--full`` and ``-c`` ?

Comment (mloubout) [reply .travis.yml]: Yes, two different things, `full` is to run all operators (born, adjoint, gradient)

File: examples/seismic/acoustic/acoustic_example.py [hunk 2]
```diff
@@ -52,16 +56,24 @@ def acoustic_setup(shape=(50, 50, 50), spacing=(15.0, 15.0, 15.0),
...
     dm = np.float32(initial_vp**2 - solver.model.m.data)
     info("Applying Forward")
     rec, u, summary = solver.forward(save=full_run, autotune=autotune)
 
+    if constant:
+        # With  a new m as Constant
+        m0 = Constant(name="m", value=.25, dtype=np.float32)
+        solver.forward(save=full_run, m=m0)
+        # With a new m as a scalar value
+        solver.forward(save=full_run, m=.25)
```
Comment (mlange05) [inline examples/seismic/acoustic/acoustic_example.py]: Should the non-constant code path still be executed as well, or should this go into an `else` clause here?

File: examples/seismic/acoustic/acoustic_example.py [hunk 3]
```diff
@@ -95,6 +107,8 @@ def run(shape=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
     parser.add_argument("-dle", default="advanced",
                         choices=["noop", "advanced", "speculative"],
                         help="Devito loop engine (DSE) mode")
+    parser.add_argument("--c", default=False, dest='constant', action='store_true',
```
Comment (mlange05) [inline examples/seismic/acoustic/acoustic_example.py]: Can we make the argument signature `'-c', '--constant'` for consistency? You won't need to specify `dest='constant'` then.

File: examples/seismic/acoustic/acoustic_example.py (hunk 2 above)
Comment (mloubout) [reply examples/seismic/acoustic/acoustic_example.py]: >Should the non-constant code path still be executed as well

This is not a non-constant path.
The rest (adjoint, born and gradient) is generated for constant/non-constant according to m, this `if` only adds an extra "test" for the constant case that is done all over the place for the non-constant case making sure we runtime args works.

File: examples/seismic/acoustic/acoustic_example.py (hunk 3 above)
Comment (mloubout) [reply examples/seismic/acoustic/acoustic_example.py]: sure

File: examples/seismic/acoustic/acoustic_example.py (hunk 2 above)
Comment (mlange05) [reply examples/seismic/acoustic/acoustic_example.py]: Ok

**Comment by mloubout** [comment]
good to go?

**Comment by mlange05** [comment]
Yes, just need travis to catch up and then we can merge.

**Comment by mloubout** [comment]
Didn't see master changed, sorry.

**Comment by FabioLuporini** [comment]
> Didn't see master changed, sorry.

That was me pushing directly to ``master`` a single word change in ``PERFORMANCE.md``

**Comment by mlange05** [comment]
Ok, all good. Merging...

**Pull request merged**